linked_list/implementation.py

```diff
@@ -1,40 +1,159 @@
+                position = position.next
+            return '{}'.format(str(list))
+        else:
+            return '[]'                             #Returns a represantion of an empty list if the linked list is indeed empty
+
 
     def __len__(self):
-        pass
+        return self.count()                         #Uses the count method to return the length of the linked list
+
 
     def __iter__(self):
-        pass
+        position = self.start
+        while position is not None:
+            yield position.elem
+            position = position.next
+
 
     def __getitem__(self, index):
-        pass
+        if len(self) - 1 < int(index):
+            raise IndexError
+        
+        position = self.start
+        counter = 0
+        while counter < index:
+            position = position.next
+            counter + 1
+        return position.elem
 
-    def __add__(self, other):
-        pass
 
+    def __add__(self, other):
+        new_list = LinkedList()
+        
+        for element in self:
+            new_list.append(element)
+        
+        for element in other:
+            new_list.append(element)
+        
+        return new_list
+        
+        
     def __iadd__(self, other):
-        pass
+        for element in other:
+            self.append(element)
+        return self
 
-    def __eq__(self, other):
-        pass
 
+    def __eq__(self, other):
+        if len(self) != len(other):
+            return False
+        
+        position = self.start
+        other_position = other.start
+        counter = 0
+        while counter < len(self):
+            if position.elem != other_position.elem:
+                return False
+            else:
+                position = position.next
+                other_position = other_position.next
+                counter += 1
+        
+        return True
+    
+    
+    def __ne__(self, other):
+        return not self is other
+    
+        
     def append(self, elem):
-        pass
+        new_node = Node(elem)                       #Creating a new node to add to the linked list
+        if self.start is None:                      #Checks if the linked list is empty, and sets the new node as first
+            self.start = new_node
+            self.end = new_node
+        else:                                       #If the linked list is not empty, adds the new node to the end and connects the second last one to it
+            self.end.next = new_node
+            self.end = self.end.next
+        self.size += 1
+
+    def __repr__(self):
+        return str(self)
 
     def count(self):
-        pass
+        return self.size
+        
 
     def pop(self, index=None):
-        pass
+        if self.start is None:                      #Raise exceptions if the linked is empty or if the index specified is larger than the length of the linked list
+            raise IndexError
+        
+        if index is not None:                                  
```

Comment:
when comparing agaisnt None you can simply this by just doing.
`if index:` or  `if not index:`